app/models/manageiq/providers/vmware/cloud_manager/refresh_parser.rb

```diff
@@ -0,0 +1,112 @@
+class ManageIQ::Providers::Vmware::CloudManager::RefreshParser < ManageIQ::Providers::CloudManager::RefreshParser
+  def self.ems_inv_to_hashes(ems, inv, options = nil)
+    new(ems, inv, options).ems_inv_to_hashes
+  end
+
+  def initialize(ems, inv, options = Config::Options.new)
+    @ems                 = ems
+    @inv                 = inv
+    @data                = {}
+    @data_index          = {}
+    @options             = options
+  end
+
+  def process_collection(collection, key)
+    @data[key] ||= []
+
+    collection.each do |item|
+      uid, new_result = yield(item)
+      next if uid.nil?
+
+      @data[key] << new_result
+      @data_index.store_path(key, uid, new_result)
+    end
+  end
+
+  def ems_inv_to_hashes
+    log_header = "MIQ(#{self.class.name}.#{__method__}) Collecting data for EMS name: [#{@ems.name}] id: [#{@ems.id}]"
+
+    $log.info("#{log_header}...")
+
+    get_vms
+
+    $log.info("#{log_header}...Complete")
+
+    @data
+  end
+
+  private
+
+  def get_vms
+    process_collection(@inv[:vms], :vms) { |vm| parse_vm(vm) }
+  end
+
+  def parse_vm(vm)
+    status = vm.status
+
+    uid           = vm.id
+    name          = vm.name
+    guest_os      = vm.operating_system
+    bitness       = !(vm.operating_system =~ /64-bit/).nil? ? 64 : 32
+    cpus          = vm.cpu
+    memory_mb     = vm.memory
+    disk_capacity = vm.hard_disks.inject(0) { |sum, x| sum + x.values[0] } * 1.megabyte
+
+    vm_disks = vm.disks.all
```

Comment:
`vm.disks` is an array of key-value pairs: `[['Hard disk 1', 1231251], ['Hard disk 2', 1287316]]`. `vm.hard_disks` is a call to an extra REST endpoint that has more informations about disks (controller type...).